c++/src/Vector.cc

```diff
@@ -196,6 +246,17 @@ namespace orc {
     }
   }
 
+  uint64_t MapVectorBatch::getMemoryUsage() {
+    return ColumnVectorBatch::getMemoryUsage()
+           + static_cast<uint64_t>(offsets.capacity() * sizeof(int64_t))
+           + keys->getMemoryUsage();
```

Comment:
Argggh, this is a silly one. Thank you for catching this bug! I didn't even get a compiler warning on this one.